src/proj.h

```diff
@@ -182,27 +262,93 @@ PJ_OBSERVATION pj_apply (PJ *P, enum pj_direction direction, PJ_OBSERVATION obs)
 /* Measure internal consistency - in forward or inverse direction */
 double pj_roundtrip(PJ *P, enum pj_direction direction, int n, PJ_OBSERVATION obs);
 
-
 int pj_show_triplet (FILE *stream, const char *banner, PJ_TRIPLET point);
 
-/* Constructor for the OBSERVATION object */
+PJ *pj_pj (PJ_CONTEXT *ctx, char *definition);
+PJ *pj_pj_argv (PJ_CONTEXT *ctx, int argc, char **argv);
+
+
+/* Initializer for the OBSERVATION object */
 PJ_OBSERVATION pj_observation (
     double x, double y, double z, double t,
     double o, double p, double k,
     int id, unsigned int flags
 );
 
+#ifndef PJ_OBSERVATION_C
+extern const PJ_OBSERVATION pj_observation_error;
+extern const PJ_OBSERVATION pj_observation_null;
+#endif
+
 #ifndef TODEG
 #define TODEG(rad)  ((rad)*180.0/M_PI)
 #endif
 #ifndef TORAD
 #define TORAD(deg)  ((deg)*M_PI/180.0)
 #endif
 
-#endif /* ndef PROJECTS_H */
+
+
+
+/* This is mostly a direct copy of (parts of) the proj_api header, but with cleaned up name space */
+
+#define PJ_VERSION 493
+
+
+PJ_CONTEXT *pj_ctx (PJ *P);
+PJ_CONTEXT *pj_ctx_alloc(void);
+
+void  pj_set_ctx (PJ *, PJ_CONTEXT *);
+void  pj_ctx_free (PJ_CONTEXT *);
+int   pj_ctx_get_errno (PJ_CONTEXT *);
+void  pj_ctx_set_errno (PJ_CONTEXT *, int);
+void  pj_ctx_set_debug (PJ_CONTEXT *, int);
+void  pj_ctx_set_logger (PJ_CONTEXT *, void (*)(void *, int, const char *));
+void  pj_ctx_set_app_data (PJ_CONTEXT *, void * );
+void *pj_ctx_get_app_data (PJ_CONTEXT *);
+
+/* file reading api, like stdio */
+typedef int *PJ_FILE;
+struct pj_fileapi;
+typedef struct pj_fileapi PJ_FILEAPI;
+struct pj_fileapi {
+    PJ_FILE  (*fopen)(PJ_CONTEXT *ctx, const char *filename, const char *access);
+    size_t   (*fread)(void *buffer, size_t size, size_t nmemb, PJ_FILE *file);
+    int      (*fseek)(PJ_FILE *file, long offset, int whence);
+    long     (*ftell)(PJ_FILE *file);
+    void     (*fclose)(PJ_FILE *file);
+};
+
+void        pj_ctx_set_fileapi (PJ_CONTEXT *, PJ_FILEAPI *);
+PJ_FILEAPI *pj_ctx_get_fileapi (PJ_CONTEXT *);
+PJ_FILEAPI *pj_get_default_fileapi();
+
+PJ_FILE  pj_ctx_fopen  (PJ_CONTEXT *ctx, const char *filename, const char *access);
```

Comment:
[This](http://lists.maptools.org/pipermail/proj/2013-June/006611.html) is Franks introduction to the context file API. He mentions that it is used in the init and gridshift file accessors. I am not sure if that is actually true since I am having a hard time finding said code. I am not very familiar with that part of the code base though, so it is likely that I have missed something.

I wouldn't be surprised if no one has ever used that functionality.